### PR TITLE
Permit retrieval of Set[T#Index] and Set[T] from ItemizedIndex

### PR DESCRIPTION
**Pull request by esarbe**
add a method that allows retrieval of a Set[T#Index] from an ItemizedIndex

### CODE REVIEW

File: core/src/main/scala/IndexedItemized.scala [hunk 1]
```diff
@@ -21,9 +21,11 @@ trait IndexedItemized extends Itemized {
 }
 
 trait ItemizedIndex[T <: IndexedItemized] {
-  def toIndex(c: T): T#Index
+  def toIndex(c: T): T#Index = c.index
 
-  def fromIndex(v: T#Index): Option[T]
+  def fromIndex(v: T#Index): Option[T] = indexMap.get(v)
+
+  val indexMap: Map[T#Index, T]
```
Comment (utaal) [inline core/src/main/scala/IndexedItemized.scala:28]: @esarbe, what do you think of something like this? You can get `.keys`, `.values` and `.size` from the map.

Comment (esarbe) [reply core/src/main/scala/IndexedItemized.scala:28]: Sure, that would work perfectly for my case!
My only "concern" is that ```fromIndex``` and ```toIndex``` seem a bit redundant now.

Comment (utaal) [reply core/src/main/scala/IndexedItemized.scala:28]: Yep, agreed. But they're "frequently accessed" I guess, so I think I'm okay with the redundancy:
 `fromIndex` and `toIndex` are the main interface, `indexMap` is for "advanced usage".

Comment (esarbe) [reply core/src/main/scala/IndexedItemized.scala:28]: Excellent. This would work for me.

**Pull request merged**